src/libs/styles.c

```diff
@@ -232,93 +269,142 @@ static void create_clicked(GtkWidget *w, gpointer user_data)
+  GtkTreeModel *model= gtk_tree_view_get_model(d->tree);
 
-  if(name)
+  GList *selected_styles = gtk_tree_selection_get_selected_rows(selection, &model);
+  GList *style_names = NULL;
+  GList *style = NULL;
+  for (style = selected_styles; style != NULL; style = style->next)
   {
-    gint res = GTK_RESPONSE_YES;
+    GValue value = {0,};
+    gtk_tree_model_get_iter(model, &iter, (GtkTreePath *)style->data);
+    gtk_tree_model_get_value(model, &iter, DT_STYLES_COL_FULLNAME, &value);
+    if(G_VALUE_HOLDS_STRING(&value))
+      style_names = g_list_append(style_names, g_strdup(g_value_get_string(&value)));
+    g_value_unset(&value);
+  }
+  g_list_free_full(selected_styles, (GDestroyNotify) gtk_tree_path_free);
 
-    if(dt_conf_get_bool("plugins/lighttable/style/ask_before_delete_style"))
-    {
-      const GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  if(style_names == NULL) return;
+
+  const gint select_cnt = g_list_length(style_names);
 
-      GtkWidget *dialog = gtk_message_dialog_new
+  gint res = GTK_RESPONSE_YES;
+
+  if(dt_conf_get_bool("plugins/lighttable/style/ask_before_delete_style"))
+  {
+    const GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+    GtkWidget *dialog = gtk_message_dialog_new
       (GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO,
-       _("do you really want to remove style '%s'?"), name);
+       ngettext("do you really want to remove %d style?", "do you realy want to remove %d styles?", select_cnt),
+       select_cnt);
 #ifdef GDK_WINDOWING_QUARTZ
-      dt_osx_disallow_fullscreen(dialog);
+    dt_osx_disallow_fullscreen(dialog);
 #endif
 
-      gtk_window_set_title(GTK_WINDOW(dialog), _("remove style?"));
-      res = gtk_dialog_run(GTK_DIALOG(dialog));
-      gtk_widget_destroy(dialog);
-    }
+    gtk_window_set_title(GTK_WINDOW(dialog), ngettext("remove style?", "remove styles?", select_cnt));
+    res = gtk_dialog_run(GTK_DIALOG(dialog));
+    gtk_widget_destroy(dialog);
+  }
 
-    if(res == GTK_RESPONSE_YES)
+  if(res == GTK_RESPONSE_YES)
+  {
+    for (style = style_names; style != NULL; style = style->next)
     {
-      dt_styles_delete_by_name(name);
-      _gui_styles_update_view(d);
+      dt_styles_delete_by_name((char*)style->data);
     }
+    _gui_styles_update_view(d);
   }
+
+  g_list_free(style_names);
 }
 
 static void export_clicked(GtkWidget *w, gpointer user_data)
 {
   dt_lib_styles_t *d = (dt_lib_styles_t *)user_data;
-  char *name = get_style_name(d);
-  if(name)
+
+  GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(d->tree));
+
+  if(gtk_tree_selection_count_selected_rows(selection) == 0) return;
+
+  GtkTreeIter iter;
+  GtkTreeModel *model= gtk_tree_view_get_model(d->tree);
+
+  GList *selected_styles = gtk_tree_selection_get_selected_rows(selection, &model);
+  GList *style_names = NULL;
+  GList *style = NULL;
+  for (style = selected_styles; style != NULL; style = style->next)
```

Comment:
for(GList *style = ...

Reply:
BTW, this very same loop is repeated in three places, probably better to put this code into a routine.

GList *_get_selected_styles(...).

You'll probably have to not   g_value_unset(&value) inside this routine and so use a g_list_free_full() for the styles_names list. See below.

Reply:
the g_value_unset is needed since the list isn't of GValue items but of gchar* items :)